### PR TITLE
fix : prebuild copy container image bugfix

### PR DESCRIPTION
**Pull request by rish2320**
# Description

In the App Configuration column , further in the PreBuild Stage the copy container image was not allowed which has been fixed now .

Fixes [#2779](https://github.com/devtron-labs/devtron/issues/2779)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
> 1: First i have added copy item and checked on dashboard that it is coming or not , then I saw it is working fine.
> 2: Secondly I have implemented css on that for proper positioning of copy item , it is also working fine.
> 3: Moreover I tested that the dropdown menu won't open on clicking the copy button . 
> 4: Finally after all this I try to copy item and pasted on my local , it is also coming . 
> 5: Also I copied the text again to see if it was working fine , it came out to be right .

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/app/list/types.tsx [hunk 1]
```diff
@@ -150,3 +150,10 @@ export interface ManifestUrlList {
     pointsTo: string
     urls: string[]
 }
+
+export interface CopyToClipboardTextProps {
+    text: string
+    rootClassName?: string
+    iconClass?: string
+    placement?: any
```
Comment (milindk8) [inline src/components/app/list/types.tsx:158]: replace 'any' by the types which takes four placement i.e bottom , top etc , get refrence from the component inside

File: src/components/CIPipelineN/TaskTypeDetailComponent.tsx [hunk 2]
```diff
@@ -207,28 +209,45 @@ export function TaskTypeDetailComponent() {
...
         return (
             <>
                 <div className="row-container mb-12">
                     <TaskFieldTippyDescription
                         taskField={TaskFieldLabel.CONTAINERIMAGEPATH}
                         contentDescription={TaskFieldDescription.CONTAINERIMAGEPATH}
                     />
-                    <div style={{ width: '80% !important' }}>
+
+                    <div className="dc__position-rel" style={{ width: '80% !important' }}>
```
Comment (shivani170) [inline src/components/CIPipelineN/TaskTypeDetailComponent.tsx:250]: Try to avoid inline css

Comment (rish2320) [reply src/components/CIPipelineN/TaskTypeDetailComponent.tsx:250]: Wasn't working inside css file.

File: src/components/app/list/types.tsx (hunk 1 above)
Comment (rish2320) [reply src/components/app/list/types.tsx:158]: Change was made by Sohel.

Comment (sohel-rp) [reply src/components/app/list/types.tsx:158]: This should have been replaced with the `Placement` type but initially was kept to `any`. We will update this.

File: src/css/base.scss [hunk 3]
```diff
@@ -274,10 +270,23 @@ a:focus {
     position: sticky;
 }
 
+.dc__top-0 {
```
Comment (shivani170) [inline src/css/base.scss:273]: Please add comment also

File: src/components/CIPipelineN/TaskTypeDetailComponent.tsx [hunk 4]
```diff
@@ -207,28 +209,45 @@ export function TaskTypeDetailComponent() {
     }
 
     const ValueContainer = (props) => {
```
Comment (shivani170) [inline src/components/CIPipelineN/TaskTypeDetailComponent.tsx:211]: Please mpve this to utils

**Comment by shivani170** [comment]
Please resolve conflict && add test cases  && issue link is missing

**Comment by rishabh12541** [comment]
> 2779



> # Description
> In the App Configuration column , further in the PreBuild Stage the copy container image was not allowed which has been fixed now .
> 
> Fixes #2779
> 
> ## Type of change
> * [x]  Bug fix (non-breaking change which fixes an issue)
> * [ ]  New feature (non-breaking change which adds functionality)
> * [ ]  Breaking change (fix or feature that would cause existing functionality to not work as expected)
> * [ ]  This change requires a documentation update
> 
> # How Has This Been Tested?
> Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
> 
> * [ ]  Test A
> * [ ]  Test B
> 
> # Checklist:
> * [x]  The title of the PR states what changed and the related issues number (used for the release note).
> * [ ]  Does this PR require documentation updates?
> * [ ]  I've updated documentation as required by this PR.
> * [x]  I have performed a self-review of my own code
> * [ ]  I have commented my code, particularly in hard-to-understand areas



> # Description
> In the App Configuration column , further in the PreBuild Stage the copy container image was not allowed which has been fixed now .
> 
> Fixes [#2779](https://github.com/devtron-labs/devtron/issues/2779)
> 
> ## Type of change
> * [x]  Bug fix (non-breaking change which fixes an issue)
> * [ ]  New feature (non-breaking change which adds functionality)
> * [ ]  Breaking change (fix or feature that would cause existing functionality to not work as expected)
> * [ ]  This change requires a documentation update
> 
> # How Has This Been Tested?
> Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
> 
> * [ ]  Test A
> * [ ]  Test B
> 
> # Checklist:
> * [x]  The title of the PR states what changed and the related issues number (used for the release note).
> * [ ]  Does this PR require documentation updates?
> * [ ]  I've updated documentation as required by this PR.
> * [x]  I have performed a self-review of my own code
> * [ ]  I have commented my code, particularly in hard-to-understand areas

File: src/css/icons.css [hunk 5]
```diff
@@ -129,6 +129,11 @@
     height: 24px;
 }
 
+.icon-dim-30 {
```
Comment (shivani170) [inline src/css/icons.css:132]: icon-dim-30 directly not working?

File: src/components/app/list/TriggerUrl.tsx [hunk 6]
```diff
@@ -156,7 +156,12 @@ export function TriggerUrlModal({ appId, envId, installedAppId, isEAMode, close
     )
 }
 
-export function CopyToClipboardText({ text, iconClass }: { text: string; iconClass?: string }) {
+export function CopyToClipboardText({
```
Comment (shivani170) [inline src/components/app/list/TriggerUrl.tsx:159]: 1. Can we extract this component out in helpers as we can reuse it
2. Rename it to more specific like  `CopyToClipboardTextWithTippy`

File: src/components/CIPipelineN/TaskTypeDetailComponent.tsx [hunk 7]
```diff
@@ -206,29 +209,31 @@ export function TaskTypeDetailComponent() {
         }
     }
 
-    const ValueContainer = (props) => {
-        let value = props.getValue()[0]?.label
+
```
Comment (shivani170) [inline src/components/CIPipelineN/TaskTypeDetailComponent.tsx:212]: Remove extra line

File: src/components/CIPipelineN/TaskTypeDetailComponent.tsx [hunk 8]
```diff
@@ -207,28 +210,45 @@ export function TaskTypeDetailComponent() {
...
+                        ) : (
+                            <span className="cn-5">Select or enter image</span>
+                        ))}
                     {React.cloneElement(props.children[1])}
                 </>
             </components.ValueContainer>
         )
     }
 
+    const MenuList = (props) => {
```
Comment (milindk8) [inline src/components/CIPipelineN/TaskTypeDetailComponent.tsx:229]: move this to utils as well

File: src/css/base.scss [hunk 9]
```diff
@@ -284,8 +288,8 @@ a:focus {
     top: 0;
 }
 
-.dc__left-35 {
```
Comment (milindk8) [inline src/css/base.scss:287]: why is it removed?

File: src/components/CIPipelineN/TaskTypeDetailComponent.tsx (hunk 8 above)
Comment (rish2320) [reply src/components/CIPipelineN/TaskTypeDetailComponent.tsx:229]: There is already a ValueContainer present in utils , so it is creating a error.

File: src/css/base.scss (hunk 9 above)
Comment (rish2320) [reply src/css/base.scss:287]: Maybe it was revolved while resolving conflicts in base.scss

Comment (rish2320) [reply src/css/base.scss:287]: Maybe it was removed while resolving conflicts in base.scss

File: src/css/base.scss [hunk 10]
```diff
@@ -284,10 +288,14 @@ a:focus {
...
```
Comment (sohel-rp) [inline src/css/base.scss:287]: dc__left-35 seems to be used at some places. Please re-check & update accordingly.

File: src/css/base.scss [hunk 11]
```diff
@@ -284,10 +288,14 @@ a:focus {
...
+.dc--left-35 {
```
Comment (sohel-rp) [inline src/css/base.scss:291]: Is this used?

**Review by sohel-rp — APPROVED** [review]

**Pull request merged**